yapapi/engine.py

```diff
@@ -762,8 +856,15 @@ async def reject_proposal(reason: str) -> events.ProposalRejected:
             return await reject_proposal("Score too low")
 
         if ignore_draft or not proposal.is_draft:
-            demand_builder = self._demand_builder
-            assert demand_builder is not None
+            assert self._demand_builder is not None
+            demand_builder = deepcopy(self._demand_builder)
```

Comment:
Do we need a `deepcopy` here and also in the Strategy?
I think we don't, so it would be better to remove one of them (probably the one in Strategy).

Reply:
hmm... probably a good catch, I didn't notice this one

Reply:
I decided to keep it - again, we don't know whether we'll always use this code in this manner and it would not be good to forget to copy when we refactor and whatnot

Reply:
> I decided to keep it - again, we don't know whether we'll always use this code in this manner and it would not be good to forget to copy when we refactor and whatnot

Yes, that's why it was used twice.

Reply:
I don't like the "we don't know what we're doing in our own library, so let's keep useless safeguards here and there" pattern. But sure, whatever.